### DISCUSSION TITLE
HomeAssistant Tasmota Integration: Unavailable Devices After HA core restart

### DISCUSSION THREAD

**Discussion by popy2k14** [arendst/Tasmota#18412] [Q&A] [answered]
I have an HASS.OS installation. It runs Mosquito broker as addon and all my 8 tasmota devices are using the official tasmota integration in HA. SetOption19 is 0 on all of them.

Zigbee2mqtt is heavily using my broker and working with no issues on ~150 zigbee devices. So my broker is ok.

Everytime when I restart HA core (NOT the broker), all my tasmota devices became unavailable and stay there.
In MQTTExplorer and the tasmota console of each device, i can see the are publishing states.
Just HA is saying "unavailable" on every light, switch, sensor of every device.

To workaround, i have to restart every tasmota device!
In this thread over at HA forum are some people with the exact same issue and some workarounds.
I have tried them all (deleting the retained discovery messages and so on), but none of them is a permanent solution,
Thread: https://community.home-assistant.io/t/tasmota-integration-unavailable-devices-after-reboot/286814/37

Any hint's what i am doing wrong?
Please help.

PS.: all those devices was running on SetOption19 1 before support was dropped.

thx

**Comment by barbudor** [comment, marked as answer]
Not using HA myself but trying to guess:

- If mosquitto doesn't restart when HA restarts, then Tasmota devices doesn't know about HA restart and doesn't republish their Discovery messages. However as these messages are retained, HA should get them when re-suscribing to the discovery topic.

- if mosquitto restart as well as HA, then Tasmota devices will see disconnections from MQTT and reconnection which will trigger republish the Discovery message

You can check in the Tasmota console if the devices see a loss of connection from MQTT and reconnection.
With WebLog3 you should also see the Discovery messages being sent in case of a reconnection.

In both cases Tasmota behavior looks correct and I would search why HA or the HAsmota plugin is not able to pickup the retained discovery messages on restart.

**Comment by popy2k14** [comment]
Thanks for clearing things up.
In my situation i am using case 1: mosquitto does'nt restart when HA does.
So i got you and think it's an issue that HA integration is'nt picking up the retained discovery messages, when it's restarted.

Will open an github issue on the HA integration side soon an post the link here, just for reference.

**Comment by popy2k14** [comment]
Just for reference, here is the issue on ha GitHub: https://github.com/home-assistant/core/issues/91462

**Comment by echo-bravo-yahoo** [comment]
I had these exact symptoms, and it turned out to be that my tasmota devices had SetOption104 set to 1 ("never send retained messages"). The discovery mechanism requires retained messages to work after restarting parts of the system. Swapping SetOption104 to 0 on all my tasmota devices resolved it for me.
